core/station/impl/src/services/external_canister.rs

```diff
@@ -319,7 +469,29 @@ impl ExternalCanisterService {
         self.external_canister_repository
             .remove(&external_canister.to_key());
 
-        // todo: remove permissions and request policies
+        // Remove all permissions related to the external canister.
+        self.permission_repository
+            .find_external_canister_call_permissions(&external_canister.canister_id)
+            .iter()
+            .for_each(|permission| {
+                self.permission_service
+                    .remove_permission(&permission.resource);
+            });
+
+        // Remove all request policies related to the external canister.
+        self.request_policy_repository
+            .find_external_canister_policies(&external_canister.canister_id)
```

Comment:
does this remove the Read policies?

Reply:
there are no read request policies, those are only permissions, but you're right we still need to remove the read and change permissions here.

Reply:
ah I confused myself 👍